<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>753</width>
    <height>590</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FreeAirTester</string>
  </property>
  <property name="whatsThis">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>20</y>
      <width>311</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Performance</string>
    </property>
    <widget class="QLabel" name="aveRpmValueLabel">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>20</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="frameShape">
      <enum>QFrame::WinPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="rpmValueLabel">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="frameShape">
      <enum>QFrame::WinPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string notr="true"/>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
    </widget>
    <widget class="QLabel" name="aveRpmLabel">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>20</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Ave. RPM:</string>
     </property>
    </widget>
    <widget class="QLabel" name="rpmLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>20</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>RPM:</string>
     </property>
    </widget>
    <widget class="QLabel" name="voltageValueLabel">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>60</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="frameShape">
      <enum>QFrame::WinPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
    </widget>
    <widget class="QLabel" name="voltageLabel_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Voltage:</string>
     </property>
    </widget>
    <widget class="QLabel" name="currentValueLabel">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>60</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="frameShape">
      <enum>QFrame::WinPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="currentLabel">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>60</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Current:</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="timeElapsedLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>510</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::WinPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="startPushButton">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>510</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopPushButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>510</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="AlternateBase">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>223</red>
          <green>238</green>
          <blue>246</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="AlternateBase">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>223</red>
          <green>238</green>
          <blue>246</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="AlternateBase">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>223</red>
          <green>238</green>
          <blue>246</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>721</width>
      <height>281</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Test</string>
     </attribute>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>10</y>
        <width>20</width>
        <height>241</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QFrame" name="autoModeFrame">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>441</width>
        <height>201</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QTableWidget" name="autoTestTable">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>321</width>
         <height>111</height>
        </rect>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QAbstractScrollArea::AdjustToContents</enum>
       </property>
       <property name="autoScroll">
        <bool>false</bool>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::CurrentChanged|QAbstractItemView::EditKeyPressed</set>
       </property>
       <property name="showDropIndicator" stdset="0">
        <bool>false</bool>
       </property>
       <property name="dragDropOverwriteMode">
        <bool>false</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectItems</enum>
       </property>
       <property name="verticalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="horizontalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="gridStyle">
        <enum>Qt::SolidLine</enum>
       </property>
       <property name="cornerButtonEnabled">
        <bool>false</bool>
       </property>
       <property name="rowCount">
        <number>3</number>
       </property>
       <property name="columnCount">
        <number>3</number>
       </property>
       <row/>
       <row/>
       <row/>
       <column>
        <property name="text">
         <string>Voltage</string>
        </property>
        <property name="font">
         <font>
          <pointsize>7</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Duty Cycle</string>
        </property>
        <property name="font">
         <font>
          <pointsize>7</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Duration</string>
        </property>
        <property name="font">
         <font>
          <family>MS Shell Dlg 2</family>
          <pointsize>7</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
       </column>
      </widget>
      <widget class="QPushButton" name="removeRowButton">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>160</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Remove Row</string>
       </property>
      </widget>
      <widget class="QPushButton" name="addRowButton">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>160</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Add Row</string>
       </property>
      </widget>
      <widget class="QComboBox" name="autoFreqComboBox">
       <property name="geometry">
        <rect>
         <x>360</x>
         <y>60</y>
         <width>69</width>
         <height>22</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>10Hz</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>100Hz</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>140Hz</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>360</x>
         <y>40</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Frequency:</string>
       </property>
      </widget>
      <widget class="QLabel" name="currentLimitLabel_2">
       <property name="geometry">
        <rect>
         <x>360</x>
         <y>100</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Current Limit:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="autoCurrentLimitSpinBox">
       <property name="geometry">
        <rect>
         <x>360</x>
         <y>120</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="focusPolicy">
        <enum>Qt::WheelFocus</enum>
       </property>
       <property name="maximum">
        <double>200.000000000000000</double>
       </property>
       <property name="value">
        <double>100.000000000000000</double>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="manualModeFrame">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>40</y>
        <width>211</width>
        <height>201</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QComboBox" name="freqComboBox">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>150</y>
         <width>69</width>
         <height>22</height>
        </rect>
       </property>
       <property name="palette">
        <palette>
         <active>
          <colorrole role="Button">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>239</red>
             <green>242</green>
             <blue>249</blue>
            </color>
           </brush>
          </colorrole>
         </active>
         <inactive>
          <colorrole role="Button">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>239</red>
             <green>242</green>
             <blue>249</blue>
            </color>
           </brush>
          </colorrole>
         </inactive>
         <disabled>
          <colorrole role="Button">
           <brush brushstyle="SolidPattern">
            <color alpha="255">
             <red>239</red>
             <green>242</green>
             <blue>249</blue>
            </color>
           </brush>
          </colorrole>
         </disabled>
        </palette>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <item>
        <property name="text">
         <string>10Hz</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>100Hz</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>140Hz</string>
        </property>
       </item>
      </widget>
      <widget class="QDoubleSpinBox" name="voltageSpinBox">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>27</y>
         <width>51</width>
         <height>22</height>
        </rect>
       </property>
       <property name="focusPolicy">
        <enum>Qt::WheelFocus</enum>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="maximum">
        <double>20.000000000000000</double>
       </property>
       <property name="value">
        <double>13.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="currentLimitLabel">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>70</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Current Limit:</string>
       </property>
      </widget>
      <widget class="QLabel" name="dutyCycleLabel">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>110</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Duty Cycle:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="currentLimitSpinBox">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>70</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="focusPolicy">
        <enum>Qt::WheelFocus</enum>
       </property>
       <property name="maximum">
        <double>200.000000000000000</double>
       </property>
       <property name="value">
        <double>100.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="voltageLabel">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Voltage:</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="dutyCycleSpinBox">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>110</y>
         <width>51</width>
         <height>22</height>
        </rect>
       </property>
       <property name="focusPolicy">
        <enum>Qt::WheelFocus</enum>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>30</number>
       </property>
      </widget>
      <widget class="QLabel" name="frequencyLabel">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>150</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Frequency:</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Settings</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>40</y>
        <width>601</width>
        <height>101</height>
       </rect>
      </property>
      <property name="title">
       <string>Test Settings</string>
      </property>
      <widget class="QLabel" name="samplesToAvgLabel">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>30</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Samples to Average:</string>
       </property>
      </widget>
      <widget class="QLabel" name="bladeCountLabel">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>91</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Fan Blade Count:</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="measurementIntervalSpinBox">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>70</y>
         <width>42</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>20000</number>
       </property>
       <property name="value">
        <number>300</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="bladeCountSpinBox">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>42</width>
         <height>22</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::UpDownArrows</enum>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="samplesToAvgSpinBox">
       <property name="geometry">
        <rect>
         <x>360</x>
         <y>30</y>
         <width>42</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>20</number>
       </property>
      </widget>
      <widget class="QLabel" name="measurementIntervalLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>141</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Measurement Interval [ms]:</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>120</y>
      <width>120</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Mode</string>
    </property>
    <widget class="QRadioButton" name="manualRadioButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>82</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Manual Mode</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="autoRadioButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>82</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Auto Mode</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>753</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSettings"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>addRowButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addTableRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>456</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>646</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeRowButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeTableRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>456</x>
     <y>387</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>389</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>enableAutoMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>647</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>591</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manualRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>enableManualMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>675</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>startTest()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>555</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>546</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopTest()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>580</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>670</x>
     <y>552</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>measurementIntervalSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setPerformanceTimerInterval(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>379</y>
    </hint>
    <hint type="destinationlabel">
     <x>175</x>
     <y>527</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>showSettingsDialog()</signal>
  <slot>startPowerSupply()</slot>
  <slot>stopPowerSupply()</slot>
  <slot>setVoltage(QString)</slot>
  <slot>setFreq(QString)</slot>
  <slot>setDutyCycle(QString)</slot>
  <slot>sendPwmCommand()</slot>
  <slot>startTest()</slot>
  <slot>stopTest()</slot>
  <slot>updateRpmDisplay()</slot>
  <slot>setBladeCount(int)</slot>
  <slot>setSampleLimit(int)</slot>
  <slot>addTableRow()</slot>
  <slot>removeTableRow()</slot>
  <slot>enableAutoMode()</slot>
  <slot>enableManualMode()</slot>
  <slot>setPerformanceTimerInterval(int)</slot>
 </slots>
</ui>
